prometheus/exporter/intervalpush/exporter.go

```diff
@@ -23,20 +23,56 @@ type exporter struct {
 }
 
 func NewExporter(url string, interval time.Duration) exp.Exporter {
-	if interval == 0 {
-		interval = defaultInterval
+	return NewExporterWithOptions(
+		url,
+		WithInterval(interval),
+		WithGrouping("instance", os.Getenv("UNIT_NAME")),
+	)
+}
+
+type Option func(exporter *exporter)
+
+func WithInterval(interval time.Duration) Option {
+	return func(e *exporter) {
+		e.t = time.NewTicker(interval)
+	}
```

Comment:
This will leak a goroutine if the ticker has a previous value :)

WDYT of using [reset](https://golang.org/pkg/time/#Ticker.Reset) here ?